### PR TITLE
Optimize batch rendering of Sprites in case of color changes

### PR DESCRIPTION
**Pull request by caryliu1999**
Re: cocos-creator/2d-tasks#381

Changes:
For batch rendering of sprite in case of color changes, update the color only when useColor is set to true.

### CODE REVIEW

File: cocos2d/core/renderer/webgl/assemblers/sprite/bar-filled.js [hunk 1]
```diff
@@ -224,17 +225,19 @@ module.exports = {
 
         let data = sprite._renderData._data,
             node = sprite.node,
+            color = node._color._val,
             matrix = node._worldMatrix,
             a = matrix.m00, b = matrix.m01, c = matrix.m04, d = matrix.m05,
             tx = matrix.m12, ty = matrix.m13;
     
-        let buffer = renderer._quadBuffer,
+        let buffer = renderer.getBuffer('quad', vfmtPosUvColor),
```
Comment (pandamicro) [inline cocos2d/core/renderer/webgl/assemblers/sprite/bar-filled.js]: 将 render component walker 中的 defaultVertexFormat 改为 vfmtPosUvColor

Comment (pandamicro) [reply cocos2d/core/renderer/webgl/assemblers/sprite/bar-filled.js]: 还要检查所有使用 _quadBuffer 和 _meshBuffer 的地方，如果没有 color attribute，要更改获取 buffer 的方式

**Review by pandamicro — CHANGES_REQUESTED** [review]

**Comment by caryliu1999** [comment]
The defaultVertexFormat in render-component-walker has been changed to vfmtPosUvColor to reduce duplicate definitions.

File: cocos2d/core/renderer/webgl/assemblers/label/bmfont.js [hunk 2]
```diff
@@ -41,7 +42,7 @@ module.exports = js.addon({
             a = matrix.m00, b = matrix.m01, c = matrix.m04, d = matrix.m05, 
             tx = matrix.m12, ty = matrix.m13;
     
-        let buffer = renderer._quadBuffer,
+        let buffer = renderer.getBuffer('quad', vfmtPosUv),
```
Comment (pandamicro) [inline cocos2d/core/renderer/webgl/assemblers/label/bmfont.js]: bmfont 也需要带上 color，这是最容易产生变色的渲染组件

File: cocos2d/core/renderer/webgl/assemblers/label/ttf.js [hunk 3]
```diff
@@ -53,7 +54,7 @@ module.exports = js.addon({
             a = matrix.m00, b = matrix.m01, c = matrix.m04, d = matrix.m05,
             tx = matrix.m12, ty = matrix.m13;
     
-        let buffer = renderer._quadBuffer,
+        let buffer = renderer.getBuffer('quad', vfmtPosUv),
```
Comment (pandamicro) [inline cocos2d/core/renderer/webgl/assemblers/label/ttf.js]: 同上

**Review by pandamicro — CHANGES_REQUESTED** [review]

**Comment by caryliu1999** [comment]
Label's color settings have been modified, now BMFont can be batch render when the color changes.

**Review by pandamicro — APPROVED** [review]

**Pull request merged**